### PR TITLE
Add codecov integration

### PR DESCRIPTION
**Pull request by tfiedor**
This PR adds codecov information, so we have information about coverage.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,7 @@
 [![devel: Ubuntu (build-&-test)](https://github.com/VeriFIT/mata/actions/workflows/ubuntu.yml/badge.svg?branch=devel)](https://github.com/VeriFIT/mata/actions/workflows/ubuntu.yml)
 [![devel: MacOS (build-&-test)](https://github.com/VeriFIT/mata/actions/workflows/macos.yml/badge.svg?branch=devel)](https://github.com/VeriFIT/mata/actions/workflows/macos.yml)
 
+[![codecov](https://codecov.io/gh/VeriFIT/mata/branch/devel/graph/badge.svg?token=9VAVD19N4D)](https://codecov.io/gh/VeriFIT/mata)
 Mata is an open source automata library that offers interface for different kinds of automata (NFA,
```
Comment (Adda0) [inline README.md]: Fix formatting of coverage badge.
```suggestion
[![codecov](https://codecov.io/gh/VeriFIT/mata/branch/devel/graph/badge.svg?token=9VAVD19N4D)](https://codecov.io/gh/VeriFIT/mata)

Mata is an open source automata library that offers interface for different kinds of automata (NFA,
```

**Review by Adda0 — APPROVED** [review]
Seems good to me. It looks like it is working.

**Comment by Adda0** [comment]
We should support running coverage locally and add a guide on how to run that in project README or somewhere similar (and reference it in project README).

**Comment by tfiedor** [comment]
> We should support running coverage locally and add a guide on how to run that in project README or somewhere similar (and reference it in project README).

I believe it worked in past, but I'm not sure if right now. I will check it out. If you want to already merge this, go for it and we can solve local coverage in separate PR.

**Comment by Adda0** [comment]
There is no rush for this PR. It does not affect anything else. Could you check whether the local coverage tests still work first? Then, we can decide whether to merge now or fix the local merge.

**Comment by tfiedor** [comment]
> There is no rush for this PR. It does not affect anything else. Could you check whether the local coverage tests still work first? Then, we can decide whether to merge now or fix the local merge.

Unfortunately, I have some issues running the coverage on my local environment, since it reports some version mismatch for `.gcda` files. There is some `coverage_report.sh` (but old, and it might be removed by summer cleaning), but still it does not work for me. I think it will be better to adress this in separate PR and build proper local coverage. The remote coverate is IMO fine, and codecov has nice UI.

**Comment by Adda0** [comment]
Thank you for checking it out. I agree. I will merge the PR momentarily. Local coverage support will be added in a new PR.

**Pull request merged**